File: app/src/main/java/com/rabtman/acgclub/base/AppConfig.java
package com.rabtman.acgclub.base;

import android.app.Application.ActivityLifecycleCallbacks;
import android.content.Context;
import com.rabtman.acgclub.BuildConfig;
import com.rabtman.acgclub.R;
import com.rabtman.acgclub.api.AcgService;
import com.rabtman.common.base.CommonApplicationLike.Lifecycle;
import com.rabtman.common.di.module.GlobeConfigModule.Builder;
import com.rabtman.common.integration.ConfigModule;
import com.rabtman.common.integration.IRepositoryManager;
import java.util.List;
import okhttp3.logging.HttpLoggingInterceptor;
import okhttp3.logging.HttpLoggingInterceptor.Level;

/**
 * @author Rabtman
 */

public class AppConfig implements ConfigModule {

  @Override
  public void applyOptions(Context context, Builder builder) {
    builder.statusBarColor(R.color.colorPrimary)
        .statusBarAlpha(0);
    if (BuildConfig.APP_DEBUG) {
      builder.addInterceptor(new HttpLoggingInterceptor().setLevel(Level.BASIC));
    }
  }

  @Override
  public void registerComponents(Context context, IRepositoryManager repositoryManager) {
    repositoryManager.injectRetrofitService(AcgService.class);
  }

  @Override
  public void injectAppLifecycle(Context context, List<Lifecycle> lifecycles) {

  }

  @Override
  public void injectActivityLifecycle(Context context,
      List<ActivityLifecycleCallbacks> lifecycles) {
  }
}

File: component-acgcomic/src/main/runalone/java/com/rabtman/acgcomic/AppConfig.kt
package com.rabtman.acgcomic

import android.app.Application.ActivityLifecycleCallbacks
import android.content.Context
import com.rabtman.acgcomic.api.AcgComicService
import com.rabtman.common.base.CommonApplicationLike.Lifecycle
import com.rabtman.common.di.module.GlobeConfigModule.Builder
import com.rabtman.common.integration.ConfigModule
import com.rabtman.common.integration.IRepositoryManager
import okhttp3.logging.HttpLoggingInterceptor
import okhttp3.logging.HttpLoggingInterceptor.Level

/**
 * @author Rabtman
 */

class AppConfig : ConfigModule {

    override fun applyOptions(context: Context, builder: Builder) {
        builder.addInterceptor(HttpLoggingInterceptor().setLevel(Level.BODY))

    }

    override fun registerComponents(context: Context, repositoryManager: IRepositoryManager) {
        repositoryManager.injectRetrofitService(AcgComicService::class.java)
    }

    override fun injectAppLifecycle(context: Context, lifecycles: List<Lifecycle>) {

    }

    override fun injectActivityLifecycle(context: Context,
                                         lifecycles: List<ActivityLifecycleCallbacks>) {
    }
}
